### PR TITLE
Upgrade to TF SDKv2

### LINKED ISSUE #12087: Upgrade to Terraform Plugin SDK v2
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

_NOTE: The work on Terraform Plugin SDK version 2 is ongoing. This issue is a placeholder for any dependencies._

### Required Changes compatible with V1 of the SDK
- [x] Remove `Partial()`/`SetPartial` usage #12083
- [x] DisableBinaryDriver for unsupported tests, mainly checks using TypeSet hashes
- [x] Develop and use alternative techniques for validating the state of `TypeSet` elements #13556 
- [x] Enable TF_SCHEMA_PANIC_ON_ERROR (default is on in V2) and identify panics/bugs. V2 of the SDK removes this env var and panics if TF_ACC is enabled #13312
- [x] Move file fixtures to child directory of `./aws` ie `./aws/files/mysql-5-6-xtrabackup.tar.gz` #13308
- [x] Ensure use of `NonRetryableError`/`RetryableError` passes non-nil #14104 
- [x] Switch to newly named validation functions #11844
- [x] Remove PromoteSingle usage
- [x] Ensure TypeMap only has Elem of primitive homogenous type (no *schema.Resource).
- [x] Remove UnsafeSetFieldRaw usage
- [x] Use custom provider name workaround for multi account/region tests #14312 

### Required Changes when switching to V2

- [x] Search and replace now broken signatures/types, this includes `terraform.ResourceProvider` `terraform.ResourceProviderFactory` and making the first argument of `CustomizeDiffFuncs` a context. #14432 
- [x] Ensure all tests and provider are compatible with the V2 binary test "process model". (Use TypeSet helpers, and be careful with concurrent design) #14067 #14381 
- [x] Remove "removed" attributes from schema as that feature has been cut. If the move to V2 SDK is a major version bump and you are ready to remove deprecated attributes, they should now go straight to deletion. #14001 

### Optional Changes

- [x] Remove usage of Exists (this is only deprecated in V2) #9953 

Linter rules:

- [x] #11869
- [x] #11870
- [x] #11871

### PR DESCRIPTION
**Pull request by appilon**
At long at last. Please note importing V1 of the SDK is still present for some helper packages in V1 (`encrypt`, `hashcode`, etc). The official recommendation is to internalize these packages or find better maintained third party libraries. I will follow up with internalizing these dependencies.

The two go modules can be imported side by side cleanly because the helper packages are "leaf" packages and don't intermix with the types throughout the SDK codebase. HOWEVER, be wary that your editor may choose to import V1 of the SDK when creating a new resource. This will result in an odd compilation error with something along the lines of `cannot use type *schema.Provider as *schema.Provider`, the reason of course is the `provider.go` file has been migrated to V2 SDK and is expecting V2 SDK namespaced types.

This annoyance will go away once the V1 dependency is removed

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closed #12087 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

No significant error rate increase or runtime increase!

### CODE REVIEW

**Comment by appilon** [comment]
@bflad I seem to have upset the linter, any chance I can get an assist on making that happy? I thought upgrading the import path to V2 SDK would be all I need.

**Comment by bflad** [comment]
@appilon my presumption is that tfproviderlint (used by awsproviderlint) will need to be updated to support the new module paths (github.com/hashicorp/terraform-plugin-sdk/... versus github.com/hashicorp/terraform-plugin-sdk/v2/...). Created a tracking issue here: https://github.com/bflad/tfproviderlint/issues/178

**Comment by bflad** [comment]
tfproviderlint v0.15.0 has been merged with SDK v2 support -- hopefully a rebase will resolve that particular issue. 🤞

File: .hashibot.hcl [hunk 1]
```diff
@@ -35,6 +35,30 @@ behavior "deprecated_import_commenter" "hashicorp_terraform" {
   EOF
 }
 
+behavior "deprecated_import_commenter" "sdkv1" {
+  import_regexp = "github.com/hashicorp/terraform-plugin-sdk/helper/(schema|resource|acctest)"
```
Comment (bflad) [inline .hashibot.hcl:39]: Let's go ahead and capture all the import paths that can be migrated:

```suggestion
  import_regexp = "github.com/hashicorp/terraform-plugin-sdk/(helper/(acctest|customdiff|logging|resource|schema|structure|validation)|terraform)"
```

File: .hashibot.hcl [hunk 2]
```diff
@@ -35,6 +35,30 @@ behavior "deprecated_import_commenter" "hashicorp_terraform" {
...
+  marker_label  = "terraform-plugin-sdk-v1"
```
Comment (bflad) [inline .hashibot.hcl:40]: Can you please add this new label in `infrastructure/repository/labels-workflow.tf` (using same color as the other label)?

File: .hashibot.hcl [hunk 3]
```diff
@@ -35,6 +35,30 @@ behavior "deprecated_import_commenter" "hashicorp_terraform" {
...
+
+  message = <<-EOF
+    Hello, and thank you for your contribution!
+
+    This project recently upgraded to [V2 of the Terraform Plugin SDK](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html)
+
+    This pull request appears to include the V1 import path of the SDK. Please import the V2 path `github.com/hashicorp/terraform-plugin-sdk/v2/helper/pkg`
```
Comment (bflad) [inline .hashibot.hcl:47]: Let's include the failing import path here for searchability:

```suggestion
    This pull request appears to include at least one V1 import path of the SDK (`${var.import_path}`). Please import the V2 path `github.com/hashicorp/terraform-plugin-sdk/v2/helper/PACKAGE`
```

**Pull request merged**